tsfresh/utilities/dataframe_functions.py

```diff
@@ -372,7 +375,55 @@ def _normalize_input_to_internal_representation(timeseries_container, column_id,
     return timeseries_container, column_id, column_kind, column_value
 
 
-def roll_time_series(df_or_dict, column_id, column_sort, column_kind, rolling_direction, max_timeshift=None):
+def _roll_out_time_series(time_shift, grouped_data, rolling_direction, max_timeshift, min_timeshift,
+                          column_sort, column_id):
+    """
+    Internal helper function for roll_time_series.
+    This function has the task to extract the rolled forecast data frame of the number `time_shift`.
+    This means it is `time_shift` in the future - starting counting from the first row
+    for each id (and kind).
+    This means we cut out the data until `time_shift`.
+    The first row we cut out is either 0 or given by the maximal allowed length of `max_timeshift`.
+    for a negative rolling direction it is reversed
+    """
+    def _f(x):
+        if rolling_direction > 0:
```

Comment:
I am a little confused, in this branch, we shift backwards in time, but the docstring below says that 

> :param rolling_direction: The sign decides, if to roll backwards or forwards in "time"

so, for negative sign, we should roll back in time and not for a positive sign as implemented here?

Reply:
No, we do actually shift forward in time.
Basically the only difference between forward and backward is the way we handle the borders and how we name the sub-time-series.

Example: if we have a window size of 3 and we shift in positive direction, it will at some point include the times [t4, t5, t6]. We would name this "t6" in this case.. If we shift backwards, it will also include this range, but we would name it "t4" in this case. 

If you image a window shifting over your time series and you move it forwards (positive), you would first only include the very first row, then the first and second, then the first second and third... So the right edge of your window increases with the `time_shift` parameters.

Reply:
Ah damn it. I have seen our confusion.
In my nomenclature, the direction is the direction of the shift of the window.
In yours, it is the data seen at each step. I will add some more docstrings here.